sdk/ai/azure-ai-projects/samples/agents/sample_agents_openapi_connection_auth.py

```diff
@@ -0,0 +1,104 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+
+"""
+FILE: sample_agents_openapi_connection_auth.py
+
+DESCRIPTION:
+    This sample demonstrates how to use agent operations with the 
+    OpenAPI tool from the Azure Agents service using a synchronous client, using
+    custom key authentication against the TripAdvisor API.
+    To learn more about OpenAPI specs, visit https://learn.microsoft.com/openapi
+
+USAGE:
+    python sample_agents_openapi_connection_auth.py
+
+    Before running the sample:
+
+    Set up an account at https://www.tripadvisor.com/developers and get an API key.
+
+    Set up a custom key connection and save the connection name following the steps at
+    https://aka.ms/azsdk/azure-ai-projects/custom-key-setup
+
+    Save that connection name as the PROJECT_OPENAPI_CONNECTION_NAME environment variable
+
+    pip install azure-ai-projects azure-identity jsonref
+
+    Set this environment variables with your own values:
+    PROJECT_CONNECTION_STRING - the Azure AI Project connection string, as found in your AI Studio Project.
+    PROJECT_OPENAPI_CONNECTION_NAME - the connection name for the OpenAPI connection authentication
+"""
+
+import os
+import jsonref
+from azure.ai.projects import AIProjectClient
+from azure.identity import DefaultAzureCredential
+from azure.ai.projects.models import OpenApiTool, OpenApiConnectionAuthDetails, OpenApiConnectionSecurityScheme 
+
+
+# Create an Azure AI Client from a connection string, copied from your AI Studio project.
+# At the moment, it should be in the format "<HostName>;<AzureSubscriptionId>;<ResourceGroup>;<HubName>"
+# Customer needs to login to Azure subscription via Azure CLI and set the environment variables
+
+project_client = AIProjectClient.from_connection_string(
+    credential=DefaultAzureCredential(),
+    conn_str=os.environ["PROJECT_CONNECTION_STRING"],
+)
+
+connection_name = os.environ["PROJECT_OPENAPI_CONNECTION_NAME"]
+conn_list = project_client.connections.list()
+conn_id = ""
```

Comment:
If you know the connection name, just call .connection.get(). No need to list all connections.

Reply:
[Updated](https://github.com/Azure/azure-sdk-for-python/pull/38820/commits/b609046c29f571c2784bd44a1148b4a0b7ebe6d2)